src/timed-event-manager.ts

```diff
@@ -1,193 +1,89 @@
-import { globalScene } from "#app/global-scene";
-import { addTextObject } from "#app/ui/text";
-import { TextStyle } from "#enums/text-style";
-import type { nil } from "#app/utils";
-import { EventType } from "#enums/event-type";
-import i18next from "i18next";
-import { GAME_HEIGHT, GAME_WIDTH } from "#app/ui-constants";
+import type { EventBanner, TimedEvent } from "#app/@types/TimedEvent";
+import { allTimedEvents } from "#app/data/all-timed-events";
+import { EventModifierType } from "#enums/event-modifier-type";
+import { isNullOrUndefined } from "#app/utils";
 
-interface EventBanner {
-  bannerKey?: string;
-  xOffset?: number;
-  yOffset?: number;
-  scale?: number;
-  availableLangs?: string[];
+function isActive(event: TimedEvent) {
+  return event.startDate < new Date() && new Date() < event.endDate;
 }
 
-interface TimedEvent extends EventBanner {
-  name: string;
-  eventType: EventType;
-  shinyMultiplier?: number;
-  friendshipMultiplier?: number;
-  startDate: Date;
-  endDate: Date;
+function isActiveOrUpcoming(event: TimedEvent) {
+  return event.startDate >= new Date() || new Date() < event.endDate;
```

Comment:
```suggestion
  return new Date() < event.endDate;
```
Should be equivalent, assuming that `startDate` is always before `endDate`

Reply:
Good point. I added a check for start date and end date in the initial filtering of the events list just to be safe rather than assume!